File: auth-service/src/main/java/com/sudo248/authservice/contronller/dto/UserDto.java
package com.sudo248.authservice.contronller.dto;

import com.sudo248.authservice.repository.entity.Gender;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDto {
    private String userId;

    private String fullName = "Soc-User";

    private String phone;

    private Date dob = new Date();

    private String bio = "";

    private String avatar = "user_default.png";

    private String cover = "user_default.png";

    private AddressDto address;

    private Gender gender = Gender.OTHER;
}

File: client/SocStaff/app/src/main/java/com/sudo248/soc_staff/data/dto/user/UserDto.kt
package com.sudo248.soc.data.dto.user

import com.sudo248.soc.domain.entity.auth.Role
import com.sudo248.soc.domain.entity.user.Gender
import java.util.Date

data class UserDto(
    val userId: String,
    val fullName: String,
    val phone: String,
    val dob: Date,
    val bio: String = "",
    val avatar: String,
    val cover: String = "",
    val address: AddressDto,
    val gender: Gender,
    val role: Role = Role.CONSUMER
)
